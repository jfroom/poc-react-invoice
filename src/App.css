.navbar.navbar-inverse {
  border-radius: 0;
  font-family: 'Roboto Slab', serif;
}
.navbar-header {
  float: left;
}
.navbar-right {
  float: right;
  margin-right: -15px;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
