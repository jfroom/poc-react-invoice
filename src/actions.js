let nextInvoiceId = 0

export const createInvoice = (title, status, items) => ({
  type: ACTION_TYPES.ADD_INVOICE,
  id: nextInvoiceId++,
  title,
  status,
  items
})

export const updateInvoice = (id, title, status, items) => ({
  type: ACTION_TYPES.UPDATE_INVOICE,
  id,
  title,
  status,
  items
})

export const deleteInvoice = (id) => ({
  type: ACTION_TYPES.DELETE_INVOICE,
  id
})

export const emailInvoice = (id) => ({
  type: ACTION_TYPES.EMAIL_INVOICE,
  id
})
