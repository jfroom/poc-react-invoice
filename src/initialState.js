const initialState = {
  invoices: {
    nextId: 4,
    invoices: [
      {
        id: 1,
        title: 'Hoyt Residence Kitchen Remodel',
        date: '1/1/2020',
        status: 'Unpaid',
        total: 7000,
        items: [
          {
            text: 'Quartz Counters',
            price: 5000,
          },
          {
            text: 'Subway Tile Backsplash',
            price: 2000,
          },
        ],
        notes: 'Thank you for your business!',
      },
      {
        id: 2,
        title: 'Lovejoy Residence Bathroom Remodel',
        date: '5/1/2017',
        status: 'Paid',
        total: 3000.99,
        items: [
          {
            text: 'Soaker tub',
            price: 1000.99,
          },
          {
            text: 'Tile work',
            price: 2000,
          },
        ],
        notes: 'Payment due net-30.',
      },
      {
        id: 3,
        title: 'Glisan Residence Living Room Remodel',
        date: '5/10/2017',
        status: 'Unpaid',
        total: 3090.99,
        items: [
          {
            text: 'Hardwood Floors',
            price: 1090.99,
          },
          {
            text: 'Ceiling Fan',
            price: 2000,
          },
        ],
        notes: 'Enjoy!',
      },
    ],
  },
}
export default initialState
