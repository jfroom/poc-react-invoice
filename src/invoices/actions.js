// @flow
import { createActions } from 'redux-actions'

export const ActionTypes = {
  ADD_INVOICE: 'ADD_INVOICE',
  EDIT_INVOICE: 'EDIT_INVOICE',
  DELETE_INVOICE: 'DELETE_INVOICE',
  CHANGE_STATUS_FILTER: 'CHANGE_STATUS_FILTER',
}

export const actions = // { addInvoice, editInvoice, deleteInvoice, changeStatusFilter }
  createActions({
    [ActionTypes.ADD_INVOICE]: invoice => (invoice),
    [ActionTypes.EDIT_INVOICE]: invoice => (invoice),
  }, ActionTypes.DELETE_INVOICE, ActionTypes.CHANGE_STATUS_FILTER)
