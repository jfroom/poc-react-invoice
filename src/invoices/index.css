/* Prefix with .App to give precedence over bootstrap styles */
.invoice-list td {
  width: 10%;
  cursor: pointer;
}
.invoice-list td:nth-child(2){
  width: 50%;
}
.invoice-list td:nth-child(2){
  width: 20%;
}

.invoice-list tr.overdue td {
  background-color: #ffd3d3;
}
.line-items .form-group {
  margin-bottom: 0;
}
.line-items td:nth-child(1) {
  width: 75%;
}
.line-items td:nth-child(3){
  width: 15%;
}
.line-items td:nth-child(3){
  width: 10%;
  vertical-align: middle;
}
.btn {
  margin: 2px;
}
.breadcrumb, .InvoiceForm h1 {
  text-transform: capitalize;
}
.InvoiceForm .line-items {
  margin-bottom: 0;
}
