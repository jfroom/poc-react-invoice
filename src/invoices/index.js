// @flow
import reducer from './reducer'

export { default as Breadcrumb } from './components/Breadcrumb'
export { default as InvoiceFormContainer } from './containers/InvoiceFormContainer'
export { default as InvoiceListContainer } from './containers/InvoiceListContainer'
export * from './actions'
export default reducer
