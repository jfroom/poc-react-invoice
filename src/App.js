import React, { Component } from 'react';
import {Navbar} from 'react-bootstrap';
import './App.css';

// HEADER
function Header() {
  return (
    <Navbar bsStyle='inverse'>
      <div className="container">
        <div className="navbar-header">
          <a className="navbar-brand" href="#">Invoice POC</a>
        </div>
        <ul className="nav navbar-nav navbar-right">
          <li>
            <a href="https://github.com/jfroom/poc-react-invoice"><small>Source</small></a>
          </li>
        </ul>
      </div>
    </Navbar>
  );
}

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <p>content here</p>
      </div>
    );
  }
}

export default App;
