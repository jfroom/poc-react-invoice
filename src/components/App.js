import React, { Component } from 'react';
import {BrowserRouter as Router, Route} from 'react-router-dom'
import { withRouter } from 'react-router'

import Header from './Header'
import Breadcrumb from './Breadcrumb'
import InvoiceListContainer from '../containers/InvoiceListContainer'
import InvoiceFormContainer from '../containers/InvoiceFormContainer'
import '../styles/App.css';

class App extends Component {
  render() {
    const BreadcrumbWithRouter = withRouter(Breadcrumb)
    return (
      <Router>
        <div>
          <Header />
          <div className='App container text-left'>
              <BreadcrumbWithRouter/>
              <Route exact path="/" component={InvoiceListContainer} />
              <Route path="/new" component={InvoiceFormContainer} />
              <Route path="/edit" component={InvoiceFormContainer} />
          </div>
        </div>
      </Router>
    );
  }
}

export default App;
