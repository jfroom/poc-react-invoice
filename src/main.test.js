import React from 'react'
import App from './main/components/App'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import rootReducer from './main/reducer'
import initialState from './initialState'
import { actions } from './invoices/actions'

let wrap
let store

beforeEach(() => {
  store = createStore(rootReducer, initialState)
  wrap = shallow(<Provider store={store}><App/></Provider>)
})

it('renders without crashing', () => {
  expect(toJson(wrap)).toMatchSnapshot()
})

it('should dispatch & reduce an action', () => {
  store.dispatch(actions.deleteInvoice(1))
  const len = initialState.invoices.invoices.length
  expect(store.getState().invoices.invoices.length).toBe(len - 1)
})
