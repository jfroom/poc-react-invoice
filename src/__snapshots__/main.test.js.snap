exports[`test renders without crashing 1`] = `<App />`;
